OsmAnd/src/net/osmand/plus/utils/AndroidUtils.java

```diff
@@ -1178,6 +1178,32 @@ private static String getStringForField(@NonNull Context ctx, @Nullable Field fi
 		return null;
 	}
 
+	public static <T> void reverseCopyOnWriteList(List<T> list) {
```

Comment:
Not needed:
Collections should be always recreated (!) in case they are changed to avoid all possible modify exceptions if they accessed from multiple threads.